measureme/src/file_serialization_sink.rs

```diff
@@ -25,17 +37,45 @@ impl SerializationSink for FileSerializationSink {
     where
         W: FnOnce(&mut [u8]),
     {
-        let mut buffer = vec![0; num_bytes];
-        write(buffer.as_mut_slice());
+        let mut data = self.data.lock();
+        let Inner {
+            ref mut file,
+            ref mut buffer,
+            ref mut buf_pos,
+            ref mut addr
+        } = *data;
 
-        let mut data = self.data.lock().expect("couldn't acquire lock");
-        let curr_addr = data.1;
-        let file = &mut data.0;
+        assert!(num_bytes <= buffer.len());
```

Comment:
If `num_bytes > buffer.len()` we could flush the buffer and then write to the file directly.

Reply:
I'll look into that.

Reply:
Implemented in https://github.com/rust-lang/measureme/pull/88/commits/b9d011168e664cd1cdf15b4f24838c4821ac6f9f.